plugin/pkg/scheduler/factory/factory.go

```diff
@@ -274,9 +276,16 @@ func (factory *ConfigFactory) makeDefaultErrorFunc(backoff *podBackoff, podQueue
 		// Note that this is extremely rudimentary and we need a more real error handling path.
 		go func() {
 			defer util.HandleCrash()
-			podID := pod.Name
+			podID := pod.Namespace + "/" + pod.Name
 			podNamespace := pod.Namespace
-			backoff.wait(podID)
+
+			entry := backoff.getEntry(podID)
+			if !entry.TryLock() {
+				glog.Warningf("Request for pod %v already in flight, abandoning", podID)
+				return
+			}
+			defer entry.Unlock()
```

Comment:
Combine the trylock/wait/unlock group into a separate function. Unlock *must* happen before podQueue.Add, or you could falsely block a subsequent error handler under really pessimistic thread ordering.